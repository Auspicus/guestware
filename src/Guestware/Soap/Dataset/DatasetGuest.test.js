import fs from 'fs'
import path from 'path'
import { DatasetGuest, DatasetGuestRow, DiffgramRowAction, SoapResponse, DatasetGuestFromResponse } from '../../../../lib/index.js'

const readMockData = async (name) => await new Promise(resolve => {
  fs.readFile(path.join(__dirname, '../../__data__', name), 'utf8', (err, data) => {
    if (err)
      throw err
    else
      resolve(data)
  })
})

describe('DatasetGuest', () => {

  test('.serialize => returns XML representation', () => {
    const dataset = new DatasetGuest([
      new DatasetGuestRow('GUEST_LOGIN', DiffgramRowAction.Inserted, {
        GuestID: '1234567890',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      })
    ])
    expect(dataset.serialize()).toMatchSnapshot()
  })

  test('.serialize => respects row ID', () => {
    const dataset = new DatasetGuest([
      new DatasetGuestRow('GUEST_COMM_METHOD', DiffgramRowAction.Inserted, {
        GuestID: '1234567890',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      }),
      new DatasetGuestRow('GUEST_COMM_METHOD', DiffgramRowAction.Inserted, {
        GuestID: '1234567890',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      })
    ])
    expect(dataset.serialize()).toMatchSnapshot()
  })

  test('.fromResponse => returns DatasetGuest from SoapResponse', async () => {
    const data = await readMockData('UpdateGuestDetailTables.xml')
    const dataset = DatasetGuestFromResponse(new SoapResponse(data))
    expect(dataset.serialize()).toMatchSnapshot()
  })

  test('.getRow => returns first DatasetGuestRow with name', () => {
    const dataset = new DatasetGuest([
      new DatasetGuestRow('GUEST_LOGIN', DiffgramRowAction.Inserted, {
        GuestID: '1234567890',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      }),
      new DatasetGuestRow('GUEST_LOGIN', DiffgramRowAction.Inserted, {
        GuestID: '0987654321',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      })
    ])
    expect(dataset.getRow('GUEST_LOGIN').properties.GuestID).toEqual('1234567890')
  })


  test('.getRowN => returns n DatasetGuestRow with name', () => {
    const dataset = new DatasetGuest([
      new DatasetGuestRow('GUEST_LOGIN', DiffgramRowAction.Inserted, {
        GuestID: '1234567890',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      }),
      new DatasetGuestRow('GUEST_LOGIN', DiffgramRowAction.Inserted, {
        GuestID: '0987654321',
        GuestLoginID: 'email@domain.com',
        GuestLoginPassword: 'password1',
      })
    ])
    expect(dataset.getRowN('GUEST_LOGIN', 1).properties.GuestID).toEqual('0987654321')
  })

})