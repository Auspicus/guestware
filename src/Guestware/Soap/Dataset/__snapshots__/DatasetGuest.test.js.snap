// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetGuest .fromResponse => returns DatasetGuest from SoapResponse 1`] = `
"
      <xs:schema
        xmlns:xs=\\"http://www.w3.org/2001/XMLSchema\\"
        xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:msprop=\\"urn:schemas-microsoft-com:xml-msprop\\"
        xmlns:mstns=\\"http://webservices.guestware.com/dstGST.xsd\\"
        id=\\"dstGST\\"
        targetNamespace=\\"http://webservices.guestware.com/dstGST.xsd\\"
        attributeFormDefault=\\"qualified\\"
        elementFormDefault=\\"qualified\\">
          <xs:element name=\\"dstGST\\" msdata:IsDataSet=\\"true\\" msdata:UseCurrentLocale=\\"true\\">
            <xs:complexType>
              <xs:choice minOccurs=\\"0\\" maxOccurs=\\"unbounded\\"/>
            </xs:complexType>
          </xs:element>
      </xs:schema>
      <diffgr:diffgram
        xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\"
        xmlns:diffgr=\\"urn:schemas-microsoft-com:xml-diffgram-v1\\">
        <dstGST xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\">
          
        <GUEST diffgr:id=\\"GUEST1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"\\">
          <GuestID>{{GuestID}}</GuestID><GstRecordType>{{GstRecordType}}</GstRecordType><NamePrefix>{{Prefix}}</NamePrefix><Surname>{{Surname}}</Surname><GivenName>{{GivenName}}</GivenName><Gender>{{Gender}}</Gender><Source>{{Source}}</Source><ProfileType>{{ProfileType}}</ProfileType><EntryDate>2018-10-12T00:24:27.023-07:00</EntryDate><EntryBy>Profile</EntryBy><LastEditDate>2018-10-12T00:24:27.023-07:00</LastEditDate><LastEditBy>Profile</LastEditBy>
        </GUEST>
      
        <GUEST_COMM_METHOD diffgr:id=\\"GUEST_COMM_METHOD1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"\\">
          <GuestID>{{GuestID}}</GuestID><GuestCommType>Email</GuestCommType><CommValue>{{Email}}</CommValue><FormatType>TEXT</FormatType><OptOut>false</OptOut><PreferredFlag>false</PreferredFlag>
        </GUEST_COMM_METHOD>
      
        <GUEST_ISSUED_ID diffgr:id=\\"GUEST_ISSUED_ID1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"\\">
          <IssueLogNo>0</IssueLogNo><GuestID>{{GuestID}}</GuestID><IssueDate>2018-10-12T07:24:26.549-07:00</IssueDate><ContactType>Email</ContactType><AssociateName>Unassigned</AssociateName><EntryBy>Profile</EntryBy><EntryDate>2018-10-12T07:24:26.549-07:00</EntryDate><LastEditBy>Profile</LastEditBy><LastEditDate>2018-10-12T07:24:26.549-07:00</LastEditDate>
        </GUEST_ISSUED_ID>
      
        <GUEST_ADDRESS diffgr:id=\\"GUEST_ADDRESS1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"\\">
          <GuestID>{{GuestID}}</GuestID><AddressType>Home</AddressType><CountryCode>null</CountryCode><PostalCode>null</PostalCode><StateCode>null</StateCode><City>null</City><AddressLine1>null</AddressLine1><AddressLine2>null</AddressLine2><Undeliverable>false</Undeliverable><PreferredFlag>true</PreferredFlag>
        </GUEST_ADDRESS>
      
        <GUEST_CONTACT_LOG diffgr:id=\\"GUEST_CONTACT_LOG1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"\\">
          <ContactLogNo>0</ContactLogNo><GuestID>{{GuestID}}</GuestID><LocationID>0</LocationID><ContactDate>2018-10-12T07:24:26.549-07:00</ContactDate><ContactTime>2018-10-12T07:24:26.549-07:00</ContactTime><ContactType>Web</ContactType><ContactItem>Profile</ContactItem><ContactStatus>Complete</ContactStatus><ContactBy>Web</ContactBy><EntryBy>Profile</EntryBy><EntryDate>2018-10-12T07:24:26.549-07:00</EntryDate><LastEditBy>Profile</LastEditBy><LastEditDate>2018-10-12T07:24:26.549-07:00</LastEditDate>
        </GUEST_CONTACT_LOG>
      
        </dstGST>
        <diffgr:before>
          
        </diffgr:before>
      </diffgr:diffgram>
    "
`;

exports[`DatasetGuest .serialize => respects row ID 1`] = `
"
      <xs:schema
        xmlns:xs=\\"http://www.w3.org/2001/XMLSchema\\"
        xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:msprop=\\"urn:schemas-microsoft-com:xml-msprop\\"
        xmlns:mstns=\\"http://webservices.guestware.com/dstGST.xsd\\"
        id=\\"dstGST\\"
        targetNamespace=\\"http://webservices.guestware.com/dstGST.xsd\\"
        attributeFormDefault=\\"qualified\\"
        elementFormDefault=\\"qualified\\">
          <xs:element name=\\"dstGST\\" msdata:IsDataSet=\\"true\\" msdata:UseCurrentLocale=\\"true\\">
            <xs:complexType>
              <xs:choice minOccurs=\\"0\\" maxOccurs=\\"unbounded\\"/>
            </xs:complexType>
          </xs:element>
      </xs:schema>
      <diffgr:diffgram
        xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\"
        xmlns:diffgr=\\"urn:schemas-microsoft-com:xml-diffgram-v1\\">
        <dstGST xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\">
          
        <GUEST_COMM_METHOD diffgr:id=\\"GUEST_COMM_METHOD1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"inserted\\">
          <GuestID>1234567890</GuestID><GuestLoginID>email@domain.com</GuestLoginID><GuestLoginPassword>password1</GuestLoginPassword>
        </GUEST_COMM_METHOD>
      
        <GUEST_COMM_METHOD diffgr:id=\\"GUEST_COMM_METHOD2\\" msdata:rowOrder=\\"1\\" diffgr:hasChanges=\\"inserted\\">
          <GuestID>1234567890</GuestID><GuestLoginID>email@domain.com</GuestLoginID><GuestLoginPassword>password1</GuestLoginPassword>
        </GUEST_COMM_METHOD>
      
        </dstGST>
        <diffgr:before>
          
        </diffgr:before>
      </diffgr:diffgram>
    "
`;

exports[`DatasetGuest .serialize => returns XML representation 1`] = `
"
      <xs:schema
        xmlns:xs=\\"http://www.w3.org/2001/XMLSchema\\"
        xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:msprop=\\"urn:schemas-microsoft-com:xml-msprop\\"
        xmlns:mstns=\\"http://webservices.guestware.com/dstGST.xsd\\"
        id=\\"dstGST\\"
        targetNamespace=\\"http://webservices.guestware.com/dstGST.xsd\\"
        attributeFormDefault=\\"qualified\\"
        elementFormDefault=\\"qualified\\">
          <xs:element name=\\"dstGST\\" msdata:IsDataSet=\\"true\\" msdata:UseCurrentLocale=\\"true\\">
            <xs:complexType>
              <xs:choice minOccurs=\\"0\\" maxOccurs=\\"unbounded\\"/>
            </xs:complexType>
          </xs:element>
      </xs:schema>
      <diffgr:diffgram
        xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\"
        xmlns:diffgr=\\"urn:schemas-microsoft-com:xml-diffgram-v1\\">
        <dstGST xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\">
          
        <GUEST_LOGIN diffgr:id=\\"GUEST_LOGIN1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"inserted\\">
          <GuestID>1234567890</GuestID><GuestLoginID>email@domain.com</GuestLoginID><GuestLoginPassword>password1</GuestLoginPassword>
        </GUEST_LOGIN>
      
        </dstGST>
        <diffgr:before>
          
        </diffgr:before>
      </diffgr:diffgram>
    "
`;
